Task3/src/Repositories/FilterApplier.hs

```diff
@@ -0,0 +1,16 @@
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module Repositories.FilterApplier 
+    ( applyFilter
+    , applyPagination) where
+import Data.Maybe (fromMaybe)
+import Data.SearchModels (SearchModel(..))
+
+applyFilter ::(SearchModel c) => (a -> b) -> (c -> Maybe b) -> (b -> b -> Bool) -> c -> [a] -> [a]
+applyFilter arrSelect searchModelSelect funcfilter searchModel arr =
+    fromMaybe arr (searchModelSelect searchModel >>= \searchValue ->
+    return $ filter (funcfilter searchValue . arrSelect) arr)
```

Comment:
It works, but let's make it a little bit more readable (+while doing so you may need to apply one more feature you haven't used yet). Try to use "let" and "where" in this function to make it more readable and easier to understand